### DISCUSSION TITLE
鸿蒙系统支持

### DISCUSSION THREAD

**Discussion by LinkRober** [Tencent/libpag#2085] [General] [answered]
有HarmonyOS版本的SDK开发计划吗

**Comment by kevingpqi123** [comment, marked as answer]
已经在计划支持了，目前还没有设备，要看微信、手Q 等产品的适配计划

**Reply by lzou** [reply]
现在适配鸿蒙的进度怎么样了？

**Reply by kevingpqi123** [reply]
已经拿到华为的设备了，预计7月份会出第一个测试版本

**Reply by zhangruoting** [reply]
鸿蒙在页面上渲染pag文件是用canvas吗？

**Reply by hyiso** [reply]
请问这个目前是什么状态呢？有没有预期发布的时间？

**Reply by wangyatong** [reply]
![image](https://github.com/user-attachments/assets/cd5f9afd-9fe8-4e8a-832a-37a149419840)
他们预计8月份发出来, 之前的帖子里面说过腾讯视频也在用， 腾讯视频已经上架next的应用市场, 这个也会提前发布出来吧

**Reply by kevingpqi123** [reply]
目前还在适配中，平台相关依赖如硬解、图片解码、底层渲染接口等已经适配差不多了，第一个版本预计在 8月上旬或中旬

**Reply by hyiso** [reply]
我们有基于 libpag 写的 Flutter 插件，使用 SurfaceTexture 的形式渲染，我看鸿蒙分支上的代码现在只有鸿蒙 ArkUI 层的使用，没看到有支持设置 surfaceId 的支持，麻烦可以考虑一下吗

**Reply by kevingpqi123** [reply]
应该会增加 PAGSurface 通过 NativeImage 或 NativeWindow创建的方式， NativeImage 相当于Android 的 SurfaceTexture，NativeWindow 相当于是 Suface ，看下这里能否满足你们的需求

**Reply by hyiso** [reply]
应该可以的，非常期待 👍

**Reply by hyiso** [reply]
不过目前 Flutter 侧只能拿到 surfaceId，肯能需要 PAGSurface 内部通过 surfaceId 去创建/获取 NativeWindow

**Comment by zhangqf1314** [comment]
我们正在想移植这个到鸿蒙平台上，能加入你们一起交流吗？
